irohad/main/impl/ordering_init.cpp

```diff
@@ -53,13 +58,17 @@ namespace iroha {
     }
 
     std::shared_ptr<OrderingGate> OrderingInit::initOrderingGate(
-        std::shared_ptr<ametsuchi::PeerQuery> wsv,
+        std::shared_ptr<ametsuchi::PeerQueryFactory> peer_query_factory,
         size_t max_size,
         std::chrono::milliseconds delay_milliseconds,
-        std::shared_ptr<ametsuchi::OrderingServicePersistentState>
+        std::shared_ptr<ametsuchi::OSPersistentStateFactory>
             persistent_state,
-        std::shared_ptr<ametsuchi::BlockQuery> block_query) {
-      auto ledger_peers = wsv->getLedgerPeers();
+        std::shared_ptr<ametsuchi::BlockQueryFactory> block_query_factory) {
+      auto query = peer_query_factory->createPeerQuery();
+      if (not query) {
+        log_->error("Cannot get the peer query");
+      }
+      auto ledger_peers = query.get()->getLedgerPeers();
```

Comment:
Isn't it potential exception on dereferencing empty optional, if you could not create `query`?

Reply:
It should be boost::none in case of any problem, however any check won't be excess